software/webapp/src/test/java/org/apache/brooklyn/entity/webapp/WebAppMigrationLiveTest.java

```diff
@@ -0,0 +1,136 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.brooklyn.entity.webapp;
+
+import com.google.common.collect.ImmutableList;
+import org.apache.brooklyn.api.entity.EntitySpec;
+import org.apache.brooklyn.api.location.ProvisioningLocation;
+import org.apache.brooklyn.core.effector.Effectors;
+import org.apache.brooklyn.core.entity.Attributes;
+import org.apache.brooklyn.core.internal.BrooklynProperties;
+import org.apache.brooklyn.core.test.BrooklynAppLiveTestSupport;
+import org.apache.brooklyn.core.test.entity.LocalManagementContextForTests;
+import org.apache.brooklyn.entity.software.base.AllowsMigration;
+import org.apache.brooklyn.test.EntityTestUtils;
+import org.apache.brooklyn.test.HttpTestUtils;
+import org.apache.brooklyn.test.support.TestResourceUnavailableException;
+import org.apache.brooklyn.util.collections.MutableMap;
+import org.reflections.Reflections;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+// This test could extend AbstractEc2LiveTest but then it will not be able to be parametrized properly.
+public class WebAppMigrationLiveTest extends BrooklynAppLiveTestSupport {
+    private static final String PROVIDER = "aws-ec2";
+    private static final String REGION_NAME = "us-west-2";
+    private static final String LOCATION_SPEC = PROVIDER + (REGION_NAME == null ? "" : ":" + REGION_NAME);
+
+    private BrooklynProperties brooklynProperties;
+    private ProvisioningLocation amazonProvisioningLocation;
+
+    @Override
+    @BeforeMethod(alwaysRun = true)
+    public void setUp() throws Exception {
+        brooklynProperties = BrooklynProperties.Factory.newDefault();
+
+        // EC2 setup obtained from AbstractEc2LiveTest
+        brooklynProperties.remove("brooklyn.jclouds." + PROVIDER + ".image-description-regex");
+        brooklynProperties.remove("brooklyn.jclouds." + PROVIDER + ".image-name-regex");
+        brooklynProperties.remove("brooklyn.jclouds." + PROVIDER + ".image-id");
+        brooklynProperties.remove("brooklyn.jclouds." + PROVIDER + ".inboundPorts");
+        brooklynProperties.remove("brooklyn.jclouds." + PROVIDER + ".hardware-id");
+
+        // Also removes scriptHeader (e.g. if doing `. ~/.bashrc` and `. ~/.profile`, then that can cause "stdin: is not a tty")
+        brooklynProperties.remove("brooklyn.ssh.config.scriptHeader");
+        mgmt = new LocalManagementContextForTests(brooklynProperties);
+        amazonProvisioningLocation = (ProvisioningLocation) mgmt.getLocationRegistry().resolve(LOCATION_SPEC);
+        super.setUp();
+    }
+
+    @DataProvider(name = "webAppEntities")
+    public Object[][] webAppEntities() {
+        Object[][] dataProviderResult;
+
+        Reflections reflections = new Reflections("org.apache.brooklyn.entity.webapp");
```

Comment:
This and `getEntitiesFromClasses` is quite cute but it would be much (much!) clearer to just return a reference to the classes explicitly. i.e. `new Object[][]{TomcatServer.class}, {Jboss7Server.class}, ...}`